### ISSUE TITLE
Failing to find static files. 

### ISSUE THREAD

**Issue by carlostighe** [chris-griffin/flask-wkhtmltopdf#3]
Im having an issue with static files and generating the pdf.   
It seems directly related to this issue https://github.com/wkhtmltopdf/wkhtmltopdf/issues/2051.  
Flask's method of serving static files means that wkhtmltopdf is trying to get them from a location that doesnt exist.   
Terminal output:  
```
Loading pages (1/6)
Warning: Failed to load file:///static/css/style.css (ignore)     
Warning: Failed to load file:///static/css/media-queries.css (ignore)
Warning: Failed to load file:///static/img/logo_trans.png (ignore)
Warning: Failed to load file:///static/img/em3_trans.png (ignore)
Warning: Failed to load file:///static/js/vendors/scripts.js (ignore)
Warning: SSL error ignored
Warning: Failed to load file://www.google-analytics.com/ga.js (ignore)
Counting pages (2/6)                                               
Resolving links (4/6)                                                       
Loading headers and footers (5/6)                                           
Printing pages (6/6)
Done                                                                      
Exit with code 1 due to network error: ContentNotFoundError
```  

Theres a "simple" work around that I could use by reproducing the whole page in a new template with absolute links but this works if you only have one page and doubles the workload.   
Im wondering is there a way to set a config variable that would reference the absolute location of the static dir?

**Comment by chris-griffin** [comment]
There is not currently a config variable, but a PR with this feature is welcomed. You can also pass in context to the templates (see https://github.com/chris-griffin/flask-wkhtmltopdf/blob/master/flask_wkhtmltopdf.py#L108) so you could optionally pass in the base path of the static directory as a variable and handle that in your template. So instead of writing entirely new templates, you can modify your existing ones to use absolute paths when you pass in a particular variable that contains an absolute path.

**Comment by carlostighe** [comment]
Right so this is what i ended up doing.   
Creating a global helper in my app factory:  
`app.jinja_env.globals['static'] = helpers.static_file`  
This was a function that dependent on the context used the regular flask `url_for` or built the url itself:  
`helpers.py`:  

```python  
def static_file(filename, pdf=False, root=None):
    if(pdf):
        return "".join([root, "static/", filename])
    else:
        return url_for('static', filename = filename)  
```   
where `pdf` was a boolean i passed into the render function and `root` was `request.url_root`  

I then called this in my templates like this:  
`<script src="{{ static('js/reportFunctions.js', pdf=pdf, root=root) }}"></script>`  
  
Unfortunately the server froze when I tried building my pdfs and I have a short timeframe so Im going to have to pursue a different course of action.  
Im leaving this here as a possible help to anyone.

**Comment by carlostighe** [comment]
So Im back and Ive been looking into this further.   
So `wkhtmltopdf` is trying to load the `js` and `css` files as local files   
`file:///static/css/style.css`  
When it does this and cant find them because they are not at this address it ignores them and produces a pdf without styling or javascript but with the content.  
When I added my above helper it seemed to try to get the files but then it just hung/hanged the server, similar to whats described here - https://github.com/wkhtmltopdf/wkhtmltopdf/issues/2254.  
  
---  

So I changed my helper above to:  
```python
def static_file(filename, pdf=False, root=None):
    if(pdf):
        basedir = os.path.abspath(os.path.dirname(__file__))
        return "".join([basedir, "/static/", filename])
    else:
        return url_for('static', filename = filename)  
```  

And its now building the pdf but Im back to the problem of my d3 graph not displaying. There are no errors being thrown so Im presuming this might be because its not rendered in time.
